mlflow/gateway/config.py

```diff
@@ -36,32 +36,52 @@ class RouteType(str, Enum):
 
 
 class OpenAIConfig(BaseModel, extra=Extra.forbid):
-    openai_api_key: Optional[str] = None
+    openai_api_key: str
     openai_api_type: Optional[str] = None
     openai_api_base: Optional[str] = "https://api.openai.com/v1"
     openai_api_version: Optional[str] = None
     openai_organization: Optional[str] = None
 
+    # pylint: disable=no-self-argument
+    @validator("openai_api_key", pre=True)
+    def validate_openai_api_key(cls, value):
+        return _resolve_api_key_from_input(value)
```

Comment:
This might look repetitive, but I think is more straightforward.

Reply:
I think it's far easier to reason about than the function that I had before. +1